### PR TITLE
Allow setting env vars for dask workers

### PR DESCRIPTION
**Pull request by yuvipanda**
Remove duplicated dask-gateway config from carbonplan & openscapes
hub as well.

Ref https://github.com/2i2c-org/pilot-hubs/issues/291#issuecomment-842655717

### CODE REVIEW

**Comment by yuvipanda** [comment]
So I've already deployed this to unblock users, but I'm leaving this up for review still.  Perhaps we can merge these to quickly unblock users but do post-merge review?

**Comment by choldgraf** [comment]
This all seems good to me - my main question is documentation. Where do we document how these options are exposed to users? (or to 2i2c team?) or is this more of a "Dask Gateway" thing?

**Comment by yuvipanda** [comment]
So the standard way to discover available options in a dask-gateway cluster works here too

image

The configuration we are using is documented upstream in https://gateway.dask.org/cluster-options.html. I'm inclined to not duplicate that here. What do you think?

**Comment by choldgraf** [comment]
makes sense to me - we should probably include some kind of documentation pointers to the Dask Gateway docs but I think it makes more sense to start from the top-down and decide which pieces we wanna link out to, rather than responsively add docs like this, so +1 on not documenting it

File: config/hubs/carbonplan.cluster.yaml [hunk 1]
```diff
@@ -140,37 +140,6 @@ hubs:
...
-                        "image": options.image,
-                        "scheduler_extra_pod_annotations": extra_annotations,
-                        "worker_extra_pod_annotations": extra_annotations,
-                        "scheduler_extra_pod_labels": extra_labels,
-                        "worker_extra_pod_labels": extra_labels,
-                    }
-                return Options(
-                    Integer("worker_cores", 2, min=1, max=16, label="Worker Cores"),
-                    Float("worker_memory", 4, min=1, max=32, label="Worker Memory (GiB)"),
-                    String("image", default="carbonplan/trace-python-notebook:latest", label="Image"),
```
[the remark this answers is not in the document]
Comment (yuvipanda) [reply config/hubs/carbonplan.cluster.yaml:170]: Ah, good catch. So this default was never actually used - there was an environment variable that overrode this, making all users use the same image as the notebook container by default. I've removed this hard-coded default and left a comment explaining this.

**Review by damianavila — APPROVED** [review]
LGTM, @yuvipanda!

**Pull request merged**